### PR TITLE
Update DeprecateDomain handler to handle global domain deprecation

### PR DESCRIPTION
**Pull request by Shaddoll**
**What changed?**
Update DeprecateDomain handler to handle global domain deprecation

**Why?**

**How did you test it?**
unit test in `common/domain`

**Potential risks**

### CODE REVIEW

File: common/domain/handler.go [hunk 1]
```diff
@@ -594,21 +588,56 @@ func (d *handlerImpl) DeprecateDomain(
...
+	isGlobalDomain := getResponse.IsGlobalDomain
+	if isGlobalDomain && !d.clusterMetadata.IsMasterCluster() {
+		return errNotMasterCluster
+	}
 	getResponse.ConfigVersion = getResponse.ConfigVersion + 1
 	getResponse.Info.Status = persistence.DomainStatusDeprecated
+	lastUpdatedTime := time.Unix(0, getResponse.LastUpdatedTime)
+	now := d.timeSource.Now()
+	// Check the failover cool down time
+	if lastUpdatedTime.Add(d.config.FailoverCoolDown(getResponse.Info.Name)).After(now) {
```
Comment (yycptt) [inline common/domain/handler.go:600]: not sure if this is useful for domain deprecation... @yux0 What's your opinion on this?

File: common/domain/handler_GlobalDomainDisabled_test.go [hunk 2]
```diff
@@ -461,6 +461,79 @@ func (s *domainHandlerGlobalDomainDisabledSuite) TestUpdateGetDomain_AllAttrSet(
 	)
 }
 
+func (s *domainHandlerGlobalDomainDisabledSuite) TestDeprecateGetDomain() {
```
Comment (vytautas-karpavicius) [inline common/domain/handler_GlobalDomainDisabled_test.go:464]: Could we do some refactoring on those tests along the way? At least for the newly added ones.
Those are already very repetitive any adding more of them makes it worse.

Comment (Shaddoll) [reply common/domain/handler_GlobalDomainDisabled_test.go:464]: What do you mean by repetitive?

Comment (vytautas-karpavicius) [reply common/domain/handler_GlobalDomainDisabled_test.go:464]: All those tests share nearly identical code. They list domain properties, register it and then assert all the properties again. I think we could make some helpers here to make it more compact and easier to read. The essence of a test gets lots in all the details around it.

Comment (Shaddoll) [reply common/domain/handler_GlobalDomainDisabled_test.go:464]: Updated the unit test.

File: common/domain/handler.go (hunk 1 above)
Comment (Shaddoll) [reply common/domain/handler.go]: Ok, I think it should be removed.

**Review by vytautas-karpavicius — APPROVED** [review]

**Pull request merged**